eclipse-scout-core/src/layout/logicalgrid/LogicalGridLayoutInfo.ts

```diff
@@ -308,26 +300,80 @@ export class LogicalGridLayoutInfo implements LogicalGridLayoutInfoModel {
       }
     }
     let sumWeightX = 0;
-    for (i = 0; i < this.cols; i++) {
+    for (let i = 0; i < this.cols; i++) {
       sumWeightX += this.weightX[i];
     }
     if (sumWeightX >= 1e-6) {
       let f = 1.0 / sumWeightX;
-      for (i = 0; i < this.cols; i++) {
+      for (let i = 0; i < this.cols; i++) {
         this.weightX[i] = this.weightX[i] * f;
       }
     }
   }
 
+  /**
+   * @param cons the current grid data
+   * @param distWidth the widths to distribute to the columns. The widgets are distributed to the columns equally.
+   * @param spanWidth the cumulated width of every cell of every row visited so far in the span range [grid.x, grid.x + grid.w]
+   * @param widths the column widths that have been distributed so far when the previous rows were visited
+   * @param fixedWidths the columns with a fixed width (weightx = 0)
+   * @param calc the function that decides whether to use the newly calculated column width or the existing one
+   */
+  protected _distributeWidth<T>(cons: LogicalGridData, distWidth: number, spanWidth: number, widths: number[], fixedWidths: boolean[], calc: (equalWidth: number, width: number) => number) {
+    let hSpan = cons.gridw;
+    let equalWidth = Math.floor((distWidth + spanWidth) / hSpan);
```

Comment:
This is not a proper way to distribute equally.
Consider the 99 distributed to 10 columns.
This will lead to 9 columns being 9 wide and the last one being 18 wide.

See pie charts or #222794 for a proper algorithm (ticket has a proof attached).

Reply:
This is existing behavior, please file a ticket if you would like to have it fixed.